### PR TITLE
Add button to update the appsecret for macos test apps

### PR DESCRIPTION
**Pull request by jinquan2019**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [ ] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [x] Did you check UI tests on the sample app? They are not executed on CI.
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Add a button to update the app secret for macOS test apps
1. Need to restart the app after updating the app secret.
2. Set app secret button will be disabled when app startups with one collector mode.

App Center mode:
image

One collector mode:
image

### CODE REVIEW

**Comment by jaelim-ms** [comment]
Remove "Internal" and display current appSecret instead. We had very similar implementation for log URL before. Look at the changes that I made in this [PR](https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1356).

File: SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift [hunk 1]
```diff
@@ -48,7 +48,7 @@ class AppCenterViewController : NSViewController, NSTextFieldDelegate, NSTextVie
   override func viewDidLoad() {
     super.viewDidLoad()
     installIdLabel?.stringValue = appCenter.installId()
-    appSecretLabel?.stringValue = (UserDefaults.standard.object(forKey: kMSAppSecret) ?? appCenter.appSecret()) as! String
+    appSecretLabel?.stringValue = prodAppSecret()
```
Comment (jaelim-ms) [inline SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift]: This is an app secret which is only available for App Center. You don't need to get PROD app secret based on start up mode. `appCenter.appSecret()` should be enough.

File: SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift [hunk 2]
```diff
@@ -29,18 +29,20 @@ class AppCenterViewController : NSViewController, NSTextFieldDelegate, NSTextVie
...
-      close(self.dbFileDescriptor)
-      UserDefaults.standard.removeObserver(self, forKeyPath: kMSStorageMaxSizeKey)
+    self.dbFileSource?.cancel()
+    close(self.dbFileDescriptor)
+    UserDefaults.standard.removeObserver(self, forKeyPath: kMSStorageMaxSizeKey)
   }
 
   override func viewWillAppear() {
     setEnabledButton?.state = appCenter.isAppCenterEnabled() ? 1 : 0
+    setAppSecretButton?.isEnabled = startUpModeForCurrentSession == 1 ? false : true
```
Comment (clpolet) [inline SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift]: You can probably introduce a constant or enum as mentioned in the iOS PR to replace the `1` value.

File: SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift [hunk 3]
```diff
@@ -48,7 +48,7 @@ class AppCenterViewController : NSViewController, NSTextFieldDelegate, NSTextVie
   override func viewDidLoad() {
     super.viewDidLoad()
     installIdLabel?.stringValue = appCenter.installId()
-    appSecretLabel?.stringValue = prodAppSecret()
+    appSecretLabel?.stringValue = appCenter.appSecret()
```
Comment (jinquan2019) [inline SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift]: I think showing app secret as empty will make sense more than showing the app secret when app startups with one collector mode. So I use the `prodAppSecret` method.

File: SasquatchMac/SasquatchMacObjC/AppCenterDelegateObjC.m [hunk 4]
```diff
@@ -44,7 +45,7 @@ - (NSString *)appSecret {
 
   // TODO: Uncomment when appSecret is moved from internal to public
```
Comment (jaelim-ms) [inline SasquatchMac/SasquatchMacObjC/AppCenterDelegateObjC.m]:
```suggestion
```

File: SasquatchMac/SasquatchMacObjC/AppCenterDelegateObjC.m [hunk 5]
```diff
@@ -44,7 +45,7 @@ - (NSString *)appSecret {
...
   // return [[MSAppCenter sharedInstance] appSecret];
```
Comment (jaelim-ms) [inline SasquatchMac/SasquatchMacObjC/AppCenterDelegateObjC.m]:
```suggestion
```

**Pull request merged**